### PR TITLE
Merge <link rel="canonical" /> tags, quick fixes (#2153)

### LINKED ISSUE #2153: Duplicate canonical tags when specifying in frontmatter.head
### What version of `starlight` are you using?

0.25.2

### What version of `astro` are you using?

4.10.2

### What package manager are you using?

pnpm

### What operating system are you using?

Mac

### What browser are you using?

Chrome

### Describe the Bug

Thank you for providing an awesome Astro theme.

I've encountered an issue where specifying a canonical tag in the frontmatter.head results in two canonical tag specifications in the HTML output. 

This duplication of canonical tags can potentially cause confusion for search engines and impact SEO performance.

`Steps to reproduce`:
- Add a canonical tag specification in head field of frontmatter
- npm run dev or npm run build
- Inspect the resulting HTML

`Expected behavior`:
Only one canonical tag should be present in the HTML output.

`Actual behavior`:
Two canonical tags are present in the HTML output:
- One from the frontmatter.head specification
- Another potentially generated by the system

`Possible solutions`:
- Implement a check to prevent duplicate canonical tags

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-8dz9uz?file=src%2Fcontent%2Fdocs%2Findex.mdx

### Participation

- [X] I am willing to submit a pull request for this issue.

### PR DESCRIPTION
**Pull request by mktbsh**
#### Description

- Closes #2153 
- What does this PR change? Give us a brief description.
   - This PR modifies the hasTag function in `starlight/utils/head.ts` to include a check for link tags. Specifically, it ensures that if a link tag has rel="canonical", duplicates are not allowed.
- Did you change something visual? A before/after screenshot can be helpful.
  - nothing

### CODE REVIEW

File: packages/starlight/utils/head.ts [hunk 1]
```diff
@@ -25,6 +25,8 @@ function hasTag(head: HeadConfig, entry: HeadConfig[number]): boolean {
 			return head.some(({ tag }) => tag === 'title');
 		case 'meta':
 			return hasOneOf(head, entry, ['name', 'property', 'http-equiv']);
+		case 'link':
+			return head.some(({ attrs }) => typeof attrs.rel === 'string' && ['canonical'].includes(attrs.rel))
```
Comment (delucis) [inline packages/starlight/utils/head.ts:29]: Can’t this be simplified to this?

```suggestion
			return head.some(({ attrs }) => attrs.rel === 'canonical')
```

File: packages/starlight/utils/head.ts
Comment (delucis) [inline packages/starlight/utils/head.ts:1]: Could you also update the comment for the `hasTag()` function to reflect the changed functionality?

**Review by delucis — CHANGES_REQUESTED** [review]
Thank you for the PR @mktbsh — the tests look good! I left a couple of small comments.

File: packages/starlight/utils/head.ts (hunk 1 above)
Comment (mktbsh) [reply packages/starlight/utils/head.ts]: I've applied the suggestions in this commit. https://github.com/withastro/starlight/pull/2154/commits/9d18b01ad2cf07c72177cd63f8314b423f44fc88

File: packages/starlight/utils/head.ts
Comment (mktbsh) [reply packages/starlight/utils/head.ts:1]: Update comment for `hasTag()`. https://github.com/withastro/starlight/pull/2154/commits/d90f96eda2a2722e8cfb2b1b8ecc567924d41404

**Review by delucis — APPROVED** [review]
Thanks again for this update @mktbsh — welcome to the Starlight contributors 🎉

**Pull request merged**